### PR TITLE
Fix torch.compile links

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121824
* #121823

Fixes https://github.com/pytorch/pytorch.github.io/issues/1567

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Review by svekars — APPROVED** [review]

File: docs/source/torch.compiler_custom_backends.rst [hunk 1]
```diff
@@ -84,7 +84,7 @@ Registration serves two purposes:
 
 * You can pass a string containing your backend function's name to ``torch.compile`` instead of the function itself,
   for example, ``torch.compile(model, backend="my_compiler")``.
-* It is required for use with the `minifier <https://pytorch.org/docs/main/compile/troubleshooting.html>`__. Any generated
+* It is required for use with the `minifier <https://pytorch.org/docs/main/torch.compiler_troubleshooting.html>`__. Any generated
```
Comment (peterbell10) [inline docs/source/torch.compiler_custom_backends.rst:87]: These links inside `rst` files, should they not be using [cross-references](https://www.sphinx-doc.org/en/master/usage/referencing.html)? I presume that would avoid dead links in future.

File: docs/source/torch.rst [hunk 2]
```diff
@@ -742,7 +742,7 @@ Optimizations
 
     compile
 
-`torch.compile documentation <https://pytorch.org/docs/main/compile/index.html>`__
+`torch.compile documentation <https://pytorch.org/docs/main/torch.compiler_index.html>`__
```
Comment (peterbell10) [inline docs/source/torch.rst:745]: This is dead, should it be this page?
```suggestion
`torch.compile documentation <https://pytorch.org/docs/main/torch.compiler.html>`__
```

File: docs/source/torch.compiler_custom_backends.rst (hunk 1 above)
Comment (lezcano) [reply docs/source/torch.compiler_custom_backends.rst:87]: Agreed.  @svekars could you or anyone from Meta fix this one?

**Review by peterbell10 — APPROVED** [review]

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**